core/integration/container_test.go

```diff
@@ -4068,3 +4069,20 @@ func TestContainerImageLoadCompatibility(t *testing.T) {
 		}
 	}
 }
+
+func TestContainerWithMountedSecretMode(t *testing.T) {
+	t.Parallel()
+
+	c, ctx := connect(t)
+	t.Cleanup(func() { c.Close() })
+
+	secret := c.SetSecret("test", "secret")
+
+	ctr := c.Container().From("alpine:3.18.2").WithMountedSecret("/secret", secret, dagger.ContainerWithMountedSecretOpts{
+		Mode:     0o666,
+		Optional: true,
+	})
+
+	_, err := ctr.WithExec([]string{"sh", "-c", "test $(cat /secret) = 'secret'"}).Sync(ctx)
```

Comment:
As far as I can tell, this test doesn't actually check whether the secret was optional, or that the configured mode was respected. I think we need a test for each. You can run `stat` to see the mode, and `test -z "$(cat /secret)"` to test that it's empty.

Reply:
I fixed the test and doc, but I'm not sure about optional? How can I mount an actual optional secret?

Reply:
Yeah it seems like you can't, so I wouldn't bother including an API for it. In Buildkit this is a thing because secret names are configured in a `Dockerfile` and optionally provided by the user separately, but Dagger handles both together; an optional secret would probably just amount to passing an empty string to `SetSecret`.

Reply:
Then should I remove the option?

Reply:
Yeah, I don't think we need it. If we can't come up with a test for it, that's proof to me.

Reply:
Fixed in [e833f5d](https://github.com/dagger/dagger/pull/5707/commits/e833f5dbe4601f632c2f932b6be06a1adb3c6347)